File: docs/snippets/all/archetypes/audio_annotation.py
"""Create and log audio annotation spans."""

import rerun as rr

rr.init("rerun_example_audio_annotation", spawn=True)

rr.set_time("time", duration=0.0)
rr.log("audio/asr", rr.AudioAnnotation("hello", span=[0.00, 0.32]))
rr.log("audio/asr", rr.AudioAnnotation("world", span=[0.34, 0.72]))

File: docs/snippets/all/archetypes/audio_annotation.rs
//! Create and log audio annotation spans.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_audio_annotation")
        .spawn()?;

    rec.set_time("time", rerun::Duration::from_secs(0.0));
    rec.log(
        "audio/asr",
        &rerun::AudioAnnotation::new("hello", [0.00, 0.32]),
    )?;
    rec.log(
        "audio/asr",
        &rerun::AudioAnnotation::new("world", [0.34, 0.72]),
    )?;

    Ok(())
}
